client/Commands/EncounterCommander.ts

```diff
@@ -123,6 +126,23 @@ export class EncounterCommander {
       Combatants: this.tracker.Encounter.Combatants().length
     });
 
+    const displayPostCombatStats = CurrentSettings().TrackerView
+      .PostCombatStats;
+    if (displayPostCombatStats) {
+      this.tracker.EventLog.AddEvent(
```

Comment:
Using the Event Log is ok for this, but I agree that a Prompt is more appropriate. Let me know if you need any pointers on using the PromptProps interface and the PromptQueue.

Reply:
Agreed. Some tips here would be appreciated.

Also, with my current abstraction, each `Combatant` has a `CombatTimer`, and so does `EncounterFlow`. I'm currently producing the output strings in those classes, so I'm not sure how to move that info to a Prompt in an efficient way. Some sort of intermediate class accepting the `Combatant` list and the `EncounterFlow`? 

Maybe my whole design is a bit off the mark here.

Reply:
You're definitely on the right track. Here's how to use the PromptProps class.
```
export interface PromptProps<T> {
  onSubmit: (T) => boolean;
  children: React.ReactChild;
  autoFocusSelector: string;
  initialValues: T;
}
```
The type `T` here represents the shape of the data you're asking the user for. Since this is a simple message prompt, that will just be `{}`. The main thing is `children`, which should be a React node containing the content of your prompt.

1. Write a function that takes your raw timer data, and returns a `PromptProps<{}>` which includes those react children. The signature should look something like this:

```
export function CombatStatsPrompt(
  totalElapsedRounds: number,
  combatants: {
    elapsedRounds: number;
    elapsedSeconds: number;
  }[]
): PromptProps<{}> {
  //get ye prompt
}
```

You can encapsulate *all* of your string logic in this method that returns the `PromptProps`. You then pass the returned prompt to `tracker.PromptQueue.Add()`.

Reply:
ApplyDamagePrompt.tsx probably contains the simplest example of a PromptProps producer.

Reply:
Here's a draft. It's functional, but I'd happily take any feedback on the UI, or the style of the React node. It feels a bit clunky to me.

I included some "DM" stats, with timing for everything that isn't a player. These will be inaccurate under edge cases where a PC is removed from the tracker before the end of combat, but I think the cost/benefit on accounting for that is more than it's worth.